### PR TITLE
Rest framework conversion

### PR DESCRIPTION
**Pull request by opxz7148**
Reopen after PR close cause by renaming

### CODE REVIEW

**Comment by opxz7148** [comment]
@TeerapatTrepopsakulsin You haven't change url reverse inside test pls change it and modify test case accordingly. Also there are join handler that haven't convert to REST yet. I've plan to do it myself tmr but I change my mind to go to health activity. Since you have time tomorrow pls convert that to rest too, by make class ActivityDetail handle POST request. Also don't forget to merge main and resolve conflict.

**Comment by TeerapatTrepopsakulsin** [comment]
- Change the URLs of the view to the new ones that use the REST framework.
- Request detail
    - Index: GET - Show list of activities, POST - Create activity
    - Detail: GET - Show activity detail, POST - Join activity, PUT - Edit activity

- Revise the tests to use new URLs.

**Comment by Thanawas-Sirilertsathit** [comment]
image
image
Does these backend stuffs work with edit page? I got error 404.

**Comment by OmegaOoh** [comment]
> image image Does these backend stuffs work with edit page? I got error 404.

I have this problem too.

File: activities/old_views.py
Comment (OmegaOoh) [inline activities/old_views.py:1]: Dead Code. Delete it.

File: activities/tests/test_create_activity.py
Comment (OmegaOoh) [inline activities/tests/test_create_activity.py:1]: test participants raise an error:
`django.urls.exceptions.NoReverseMatch: Reverse for 'join' not found. 'join' is not a valid view function or pattern name.`

File: mysite/urls.py
Comment (OmegaOoh) [inline mysite/urls.py:1]: explains this change

File: activities/views/util.py
Comment (OmegaOoh) [inline activities/views/util.py:1]: Doesn't this need to be convert to REST?

**Review by OmegaOoh — CHANGES_REQUESTED** [review]

**Comment by OmegaOoh** [comment]
Another thing is, before the next review request please, merge the code from the main. I can't let you merge without reviewing it.

File: mysite/urls.py
Comment (opxz7148) [reply mysite/urls.py:1]: Rest framwork has different default format. Need to set it to match with our format.

**Review by OmegaOoh — APPROVED** [review]
Looking good

**Review by OmegaOoh — CHANGES_REQUESTED** [review]
You need to merge from main before actually merge tho

File: mysite/urls.py
Comment (opxz7148) [reply mysite/urls.py:1]: Oh my eyes are nut. I thought you comment on upper one. That thing is unnecessary will remove. 
Screenshot 2567-10-19 at 20 08 55

Comment (opxz7148) [reply mysite/urls.py:1]: Remove it on #[0cb0220](https://github.com/OmegaOoh/ku-tangtee/pull/84/commits/0cb02202a53f0bb14114e35b51b0f6d5d7d08084)

**Review by Thanawas-Sirilertsathit — APPROVED** [review]
The refresh button now appear

**Review by OmegaOoh — APPROVED** [review]

**Pull request merged**